Add Interactive brokered authentication and wam content (#42279)
dotnet/docs: docs/azure/sdk/authentication/additional-methods.md

### Enable applications for interactive browser authentication

Perform the following steps to enable the application to authenticate through the interactive browser flow. These steps also work for the [device code authentication](#device-code-authentication) flow described later. This process is only necessary if you are using `InteractiveBrowserCredential` in your code.

1. On the [Azure portal](https://portal.azure.com), navigate to Microsoft Entra ID and select **App registrations** on the left navigation.
1. Select the registration for your app, then select **Authentication**.
1. Under **Advanced settings**, select **Yes** for **Allow public client flows**.
1. Select **Save** to apply the changes.
1. To authorize the application for specific resources, navigate to the resource in question, select **API Permissions**, and enable **Microsoft Graph** and other resources you want to access. Microsoft Graph is usually enabled by default.

    > [!IMPORTANT]
    > You must also be the admin of your tenant to grant consent to your application when you sign in for the first time.

> If you can't configure the device code flow option, your application might need to be multitenant. To make this change, navigate to the **Authentication** panel, select **Accounts in any organizational directory** (under **Supported account types**), and then select **Yes** for **Allow public client flows**.

Reviewer: This paragraph about device code flow option seems out of place - to which device code option are we referring?

Author: Removed

For more exact control, such as setting redirect URIs, you can supply specific arguments to `InteractiveBrowserCredential` such as `redirect_uri`.

> ### Authenticate the default system account via WAM

Reviewer 2: Let's stick with the same heading Python has.
```suggestion
## Interactive brokered authentication
```

> Web Account Manager (WAM) is a system authentication broker service that allows apps to seamlessly request OAuth tokens from identity providers, such as Microsoft Entra ID. WAM enables identity providers to natively plug into the OS and provide the service to other apps to streamline the login process. WAM offers the following benefits:

Reviewer 2: It's important to call out right away that this feature is for Windows only. We don't yet have a solution for macOS and Linux.
```suggestion
Web Account Manager (WAM) is a system authentication broker service for Windows that allows apps to seamlessly request OAuth tokens from identity providers, such as Microsoft Entra ID. WAM enables identity providers to natively plug into the OS and provide the service to other apps to streamline the login process. WAM offers the following benefits:
```

- **Feature support**: Apps can access OS-level and service-level capabilities, including Windows Hello, conditional access policies, and FIDO keys.
- **Streamlined single sign-on**: Apps can use the built-in account picker, allowing the user to select an existing account instead of repeatedly entering the same credentials.
- **Enhanced security**: Bug fixes and enhancements ship with Windows.
- **Token protection**: Refresh tokens are device-bound, and apps can acquire device-bound access tokens.

Many people always sign in to Windows with the same user account and, therefore, only ever want to authenticate using that account. WAM also supports a silent login process that automatically uses a default account so the user does not have to repeatedly select it.

> To use WAM and the default system account in your app:

Reviewer 2: ```suggestion
To use WAM in your app:
```

1. Add the [Azure.Identity](https://www.nuget.org/packages/Azure.Identity) and [Azure.Identity.Broker](https://www.nuget.org/packages/Azure.Identity.Broker) NuGet packages to your project.

    ```dotnetcli
    dotnet add package Azure.Identity
    dotnet add package Azure.Identity.Broker
    ```

1. Get the handle of the parent window to which the WAM account picker window should be docked.

    ```csharp
    // Form1.cs
    private async void testBrokeredAuth_Click(object sender, EventArgs e)
    {
        IntPtr windowHandle = this.Handle;
    }
    ```

    > [!NOTE]
    > This example shows how to get the window handle for a Windows Forms app. Visit the [Parent window handles](/entra/msal/dotnet/acquiring-tokens/desktop-mobile/wam#parent-window-handles) and [Retrieve a window handle](/windows/apps/develop/ui-input/retrieve-hwnd) articles for more information about retrieving window context for other types of apps.

> 1. Create an instance of `InteractiveBrowserCredential` in your app. The credential requires the handle of the parent window that's requesting the authentication flow. On Windows, the handle is an integer value that uniquely identifies the window. Optionally, set the `UseDefaultBrokerAccount` option to `true` to enable silent brokered authentication, which will automatically select the default account.

Reviewer 2: ```suggestion
1. Create an instance of `InteractiveBrowserCredential` in your app. The credential requires the handle of the parent window that's requesting the authentication flow. On Windows, the handle is an integer value that uniquely identifies the window. Optionally, set the [UseDefaultBrokerAccount](/dotnet/api/azure.identity.broker.interactivebrowsercredentialbrokeroptions.usedefaultbrokeraccount#azure-identity-broker-interactivebrowsercredentialbrokeroptions-usedefaultbrokeraccount) property to `true` to enable silent, brokered authentication, which will automatically select the default system account.
```

    ```csharp
    using Azure.Identity;
    using Azure.Identity.Broker;
    
    // code omitted for brevity
    private async void testBrokeredAuth_Click(object sender, EventArgs e)
    {
        IntPtr windowHandle = this.Handle;
    
        InteractiveBrowserCredential credential = new(
            new InteractiveBrowserCredentialBrokerOptions(windowHandle)
            {
                // Enable silent brokered authentication using the default account
                UseDefaultBrokerAccount = true,
            }
        );
    }
    ```

1. Use the broker-enabled `InteractiveBrowserCredential` instance.

> Once you opt into this behavior, the credential type attempts to sign in by asking the underlying Microsoft Authentication Library (MSAL) to perform the sign-in for the default system account. If the sign-in fails, the credential type falls back to displaying the account picker dialog, from which the user can select the appropriate account.

Reviewer 2: ```suggestion
Once you opt into this behavior, the credential attempts to sign in by asking the underlying Microsoft Authentication Library (MSAL) to perform the sign-in for the default system account. If the sign-in fails, the credential type falls back to displaying the account picker dialog, from which the user can select the appropriate account.
```

For more exact control, such as setting redirect URIs, you can supply specific arguments to `InteractiveBrowserCredential` such as `redirect_uri`.

## Interactive brokered authentication

> This method interactively authenticates an application through [`InteractiveBrowserCredential`](/dotnet/api/azure.identity.interactivebrowsercredential?view=azure-dotnet) by collecting user credentials using the system authentication broker. A system authentication broker is an app running on a user's machine that manages the authentication handshakes and token maintenance for all connected accounts. Currently, only the Windows authentication broker, Web Account Manager (WAM), is supported. Users on macOS and Linux will be authenticated through a browser.

Reviewer: ```suggestion
This method interactively authenticates an application through [`InteractiveBrowserCredential`](/dotnet/api/azure.identity.interactivebrowsercredential?view=azure-dotnet) by collecting user credentials using the system authentication broker. A system authentication broker is an app running on a user's machine that manages the authentication handshakes and token maintenance for all connected accounts. Currently, only the Windows authentication broker, Web Account Manager (WAM), is supported. Users on macOS and Linux will be authenticated through the non-brokered interactive browser flow.
```

> WAM enables identity providers such as Microsoft Entra ID to natively plug into the OS and provide the service to other apps to streamline the login process. WAM offers the following benefits:

Reviewer: ```suggestion
WAM enables identity providers such as Microsoft Entra ID to natively plug into the OS and provide the service to other apps to provide a more secure login process. WAM offers the following benefits:
```

- **Feature support**: Apps can access OS-level and service-level capabilities, including Windows Hello, conditional access policies, and FIDO keys.
- **Streamlined single sign-on**: Apps can use the built-in account picker, allowing the user to select an existing account instead of repeatedly entering the same credentials.
- **Enhanced security**: Bug fixes and enhancements ship with Windows.
- **Token protection**: Refresh tokens are device-bound, and apps can acquire device-bound access tokens.

Personal Microsoft accounts and work or school accounts are supported. If a supported version of Windows is used, the default browser-based UI is replaced with a smoother authentication experience, similar to Windows built-in apps.

> Interactive brokered authentication enables the application for all operations allowed by the interactive login credentials. As a result, if you're the owner or administrator of your subscription, your code has inherent access to most resources in that subscription without having to assign any specific permissions.

Reviewer: Why is this special to brokered auth?

The following screenshot shows the user sign-in experience:

:::image type="content" source="../media/web-account-manager-sign-in-account-picker.png" alt-text="A screenshot that shows the sign-in experience when using the interactive browser broker credential to authenticate a user." :::

### Authenticate the default system account via WAM

Many people always sign in to Windows with the same user account and, therefore, only ever want to authenticate using that account. WAM and `InteractiveBrowserCredential` also support a silent login process that automatically uses a default account so the user does not have to repeatedly select it.

The following example shows how to enable sign-in with the default system account:

```csharp
using Azure.Identity;
using Azure.Identity.Broker;

// App code omitted for brevity

IntPtr windowHandle = this.Handle;

var credential = new InteractiveBrowserCredential(
    new InteractiveBrowserCredentialBrokerOptions(windowHandle)
    {
        // Enable silent brokered authentication using the default account

>         UseDefaultBrokerAccount = true,

Reviewer 2: I recommend creating a sample app to accompany this doc. Then you can import code snippets and highlight important lines of code like this one.
